From d7a2ff17b15842bf9a3de65ca1ba577bdf568e79 Mon Sep 17 00:00:00 2001
From: Wang Dong <dongdwdw@linux.vnet.ibm.com>
Date: Fri, 24 Mar 2017 03:11:08 +0100
Subject: [PATCH] parted: check the name of partition first when to name a
 partition

The previous function works well for the labels supporting naming
partition, but not for these which don't. If the disk label does not
support partition naming, two exceptions will be raised. Even after the first
error indicates it does not support name, parted yet asks user for
name in iteractive mode.

First check if the disk label supports partition naming and if it
does, it will continue; otherwise, it will stop and raise an
exception.

Signed-off-by: Wang Dong <dongdwdw@linux.vnet.ibm.com>
Signed-off-by: Hendrik Brueckner <brueckner@linux.vnet.ibm.com>
---
 parted/parted.c | 7 +++++++
 1 file changed, 7 insertions(+)

Index: parted-3.2/parted/parted.c
===================================================================
--- parted-3.2.orig/parted/parted.c
+++ parted-3.2/parted/parted.c
@@ -930,6 +930,13 @@ do_name (PedDevice** dev, PedDisk** disk
         if (!diskp)
                 goto error;
 
+        if (!ped_disk_type_check_feature((*diskp)->type, PED_DISK_TYPE_PARTITION_NAME)) {
+                ped_exception_throw (PED_EXCEPTION_ERROR, PED_EXCEPTION_CANCEL,
+                                     _("%s disk labels do not support partition name."),
+				     (*diskp)->type->name);
+                goto error;
+        }
+
         if (!command_line_get_partition (_("Partition number?"), *diskp, &part))
                 goto error;
 
